src/main/java/in/ac/bits/protocolanalyzer/persistence/repository/ESFactoryImpl.java

```diff
@@ -1,25 +1,61 @@
 package in.ac.bits.protocolanalyzer.persistence.repository;
 
+import javax.annotation.Resource;
+
 import org.elasticsearch.common.settings.ImmutableSettings;
+import org.springframework.core.env.Environment;
+import org.springframework.data.elasticsearch.repository.config.EnableElasticsearchRepositories;
 import org.elasticsearch.node.NodeBuilder;
+import org.springframework.context.annotation.PropertySource;
 import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Component;
 
 @Component
 @Scope("prototype")
+@EnableElasticsearchRepositories(basePackages = "in.ac.bits.protocolanalyzer.persistence.repository")
+@PropertySource(value = "classpath:META-INF/elasticsearch.properties")
 public class ESFactoryImpl implements ESFactory {
 
+	@Resource
+    private Environment environment;
```

Comment:
@prasadtalasila when I try to access Environment using getProperty(), it throws a NullPointerException.
I checked and found that field environment is consistently null in this case, but I could not figure out why.
Please help. I will refactor and optimize code once it is in running condition.